channeldb/reports.go

```diff
@@ -0,0 +1,332 @@
+package channeldb
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/channeldb/kvdb"
+)
+
+var (
+	// reportBucket is a bucket which holds a record of the final set
+	// of resolutions which were processed for a channel.
+	// [reportBucket]
+	//	[chainHashBucket]
+	//		[channelBucket]
+	//			[resolversBucket]
```

Comment:
How does this structure compare to grouping the reports as a sub-bucket somewhere closer to the other channel data?

Reply:
Moved this set of nested buckets under the `closedChannelBucket` top level. 
I've kept chain hash/ channel nesting (don't know why we don't have it for regular close summaries?)

As is, close summary is pretty difficult to add fields to because they summary is written as a single key-value record with a bunch of optional fields at the end. Could be nice to (one day) move channel summaries into this bucket as well?
```
[closedChannelBucket]
      [chainHashBucket]
          [channelBucket]
            [resolversBucket]
            +[closeSummary]
```